src/uri.rs

```diff
@@ -179,6 +175,39 @@ impl Uri {
     }
 }
 
+fn parse_scheme(s: &str) -> Option<usize> {
+    s.find(':')
+}
+
+fn parse_authority(s: &str) -> Option<usize> {
+    let v: Vec<&str> = s.split("://").collect();
+    let auth = v.last().unwrap()
```

Comment:
I planned on filing issues for a couple more passes through the file, removing the temporary `Vec`s and what-not. It can be fixed here if you want, or in another pass.